### ISSUE TITLE
How to keep the style after reloading page

### ISSUE THREAD

**Issue by ErikGarfia** [ckeditor/ckeditor5#8023]
## 📝 Provide a description

I have a plugin that changes both the style and html tag format from selected text, It does very well its task, when I change the html tag format my database saves the html format without any problem, when I reload the page the html format is saved but the style disappears.

## 📃 Other details

this is my code:

**editing**
```js
export class SmallCapsEditing extends Plugin {
    init() {
        const editor = this.editor;
        const schema = editor.model.schema;
        const t = editor.t;

        schema.extend('$text', {
            allowAttributes: [SMALL_CAPS, 'tooltip', 'id']
        });



        editor.commands.add(SMALL_CAPS, new SmallCapsCommand(editor, SMALL_CAPS));
    }
}
```

**UI**

```js
        editor.ui.componentFactory.add('smallCaps', locale => {
            const command = editor.commands.get(SMALL_CAPS);
            var view = new ButtonView(locale);

            view.set({
                label: 'Tooltips',
                icon: smallCapsIcon,
                tooltip: true,
                class: 'small-caps',
            });

            //this is the input text that appears when the user clicks on plugin's botton
            view.on('execute', () => {

                
                editor.conversion.attributeToElement({
                    model: SMALL_CAPS,
                    view: {
                        name: 'tooltip',
                        styles: {
                            'color': 'blue',
                        },
                        attributes: {
                            'text': prompt('Add a tooltip')
                        },

                    },
                    upcastAlso: [{
                        styles: {
                            'color': 'blue'
                        }
                    }]


                });

            });
            view.bind('isOn', 'isEnabled').to(command, 'value', 'isEnabled');

            this.listenTo(view, 'execute', () => {
                editor.execute(SMALL_CAPS)
            });
            return view;
        });

    }
}
```

**command**

```js

	/**
	 * @inheritDoc
	 */
	refresh() {
		const model = this.editor.model;
		const doc = model.document;

		this.value = this._getValueFromFirstAllowedNode();
		this.isEnabled = model.schema.checkAttributeInSelection( doc.selection, this.attributeKey );
	}

	execute( options = {} ) {
		const model = this.editor.model;
		const doc = model.document;
		const selection = doc.selection;
		const value = ( options.forceValue === undefined ) ? !this.value : options.forceValue;

		model.change( writer => {
			if ( selection.isCollapsed ) {
				if ( value ) {
					
					writer.setSelectionAttribute( this.attributeKey, true );
				} else {
					writer.removeSelectionAttribute( this.attributeKey );
				}
			} else {
				const ranges = model.schema.getValidRanges( selection.getRanges(), this.attributeKey );

				for ( const range of ranges ) {
					if ( value ) {
						writer.setAttribute( this.attributeKey, value, range );
					} else {
						writer.removeAttribute( this.attributeKey, range );
					}
				}
			}
		} );
	}

	_getValueFromFirstAllowedNode() {
		const model = this.editor.model;
		const schema = model.schema;
		const selection = model.document.selection;
		if ( selection.isCollapsed ) {
			return selection.hasAttribute( this.attributeKey );
		}

		for ( const range of selection.getRanges() ) {
			for ( const item of range.getItems() ) {
				if ( schema.checkAttribute( item, this.attributeKey ) ) {
					return item.hasAttribute( this.attributeKey );
				}
			}
		}

		return false;
	}
}
```

and my plugin does this 
<img width="800" alt="Captura de Pantalla 2020-09-03 a la(s) 13 10 06" src="https://user-images.githubusercontent.com/44913795/92151473-cc0ebb80-ede6-11ea-90c9-84597892e611.png">

and changes the format from 

`<h2>Sample</h2>`

to

`<tooltip style="color:blue;" text="hello world">Sample</tooltip>`

but when I reload the page, the html tag is saved but the style dissapears

<img width="800" alt="Captura de Pantalla 2020-09-03 a la(s) 13 12 36" src="https://user-images.githubusercontent.com/44913795/92151665-2576ea80-ede7-11ea-98b5-2ecc073c0c48.png">

is there any solution for this?

**Comment by jodator** [comment]
@ErikGarfia I think that you have one thing mixed up. Other than that your code should do it's job.

1.  The `editor.conversion.attributeToElement(` defines how model elements are converted to view. It should be defined once. Our pattern is to define this in the editing plugin ( `SmallCapsEditing`) not in the UI part.
2.  To change model, you only need to a command (you have this already).
3.  This command should be executed by the UI component on `view`'s    `'execute'` event (as you already have).

Now, why the style is not retained? Because you also need to define how view is translated to the model since [conversion](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/editing-engine.html#conversion), is two-way process. This converter must be defined _before_ the view is being read.

So, in other words move your `editor.conversion.attributeToElement({` from the UI to the `SmallCapsEditing` constructor and it should work.

ps.: I've edited your post for readability using three backticks ` ```js` instead of one backtick `` ` ``.

**Comment by ErikGarfia** [comment]
thanks for your answer @jodator 
My plugin shows a prompt when the user click on the plugins Botton, when I change `editor.conversion.attributeToElement` to the `SmallCapsEditing` this prompt appears when I reload the page, I mean, instead of appearing when the user click on the Botton, the prompt appears as soon as I reload the page

**Small Caps Editing**
``` 
export class SmallCapsEditing extends Plugin {
    init() {
        const editor = this.editor;
        const schema = editor.model.schema;
        const t = editor.t;

        schema.extend('$text', {
            allowAttributes: [SMALL_CAPS, 'tooltip', 'id']
        });
        editor.conversion.attributeToElement({
            model: SMALL_CAPS,
            view: {
                name: 'tooltip',
                styles: {
                    'color': 'blue',
                },
                attributes: {
                   //this is the prompt that appears where the user types a custom attribute
                    'text': prompt('Add a tooltip')
                },

            },
            upcastAlso: [{
                styles: {
                    'color': 'blue'
                }
            }]
        });
        editor.commands.add(SMALL_CAPS, new SmallCapsCommand(editor, SMALL_CAPS));
    }
}
```

**Small Caps UI**

```
export class SmallCapsUi extends Plugin {
    init() {
        const editor = this.editor;

        editor.ui.componentFactory.add('smallCaps', locale => {
            const command = editor.commands.get(SMALL_CAPS);
            var view = new ButtonView(locale);

            view.set({
                label: 'Tooltips',
                icon: smallCapsIcon,
                tooltip: true,
                class: 'small-caps',
            });


            view.bind('isOn', 'isEnabled').to(command, 'value', 'isEnabled');

            this.listenTo(view, 'execute', () => {
                editor.execute(SMALL_CAPS)
            });
            return view;
        });

    }
}
```

how can I avoid this? Maybe this is the wrong way to put the prompt

**Comment by Mgsy** [comment]
This is a duplicate of #7948. Let's move the discussion to the original issue.
